hail_scripts/v01/utils/elasticsearch_client.py

```diff
@@ -103,8 +103,33 @@ def export_vds_to_elasticsearch(
                 "sample_id: s",  # add sample_id to each genotype struct to keep track of
             ])
 
-            genotypes_root = "gs" if not discard_missing_genotypes else "gs.filter(g => g.isCalled())"
-            vds = vds.annotate_variants_expr("va.genotypes = %(genotypes_root)s.map(g => { %(genotype_struct_expr)s }).collect()" % locals())
+            vds = vds.annotate_variants_expr("va.genotypes = gs.map(g => { %(genotype_struct_expr)s }).collect()" % locals())
+
+            sample_fields_annotation_template = "va.samples_%(field_name)s = va.genotypes.filter(gen => %(field_filter)s).map(gen => gen.sample_id).toSet"
+            for i in range(1, 3):
+                vds = vds.annotate_variants_expr(sample_fields_annotation_template % dict(
+                    field_name='num_alt_{}'.format(i), field_filter='gen.num_alt == {}'.format(i)
+                ))
+            vds = vds.annotate_variants_expr(sample_fields_annotation_template % dict(
+                field_name='no_call', field_filter='gen.num_alt == -1'
+            ))
+
+            for i in range(5, 95, 5):
```

Comment:
The code looks fine to me. 

The number of bins still seems excessive. Looking at the search_cache in mongo, 
here's the number of searches with the filters set to a particular bin:

```
> [0, 5, 10, 15, 20, 25, 30, 35, 40, 45, 50, 55, 60, 65, 70, 75, 80, 85, 90, 95].forEach( function(i) { print(i, "<= min GQ < ", i+5, ": ", db.search_cache.find({"search_spec.genotype_quality_filter.min_gq" : {$gte: i, $lt : i +5 }}).count() ) })
931  searches with:  0 <= min GQ <  5
  4  searches with:  5 <= min GQ <  10
 82  searches with:  10 <= min GQ <  15
  5  searches with:  15 <= min GQ <  20
118  searches with:  20 <= min GQ <  25
  5  searches with:  25 <= min GQ <  30
617  searches with:  30 <= min GQ <  35
  0  searches with:  35 <= min GQ <  40
  0  searches with:  40 <= min GQ <  45
  0  searches with:  45 <= min GQ <  50
  0  searches with:  50 <= min GQ <  55
  0  searches with:  55 <= min GQ <  60
  0  searches with:  60 <= min GQ <  65
  0  searches with:  65 <= min GQ <  70
  0  searches with:  70 <= min GQ <  75
  0  searches with:  75 <= min GQ <  80
  0  searches with:  80 <= min GQ <  85
  0  searches with:  85 <= min GQ <  90
  0  searches with:  90 <= min GQ <  95
  0  searches with:  95 <= min GQ <  100
```
and

```
999  searches with:  0 <= min GQ <  5
  4  searches with:  5 <= min GQ <  10
 25  searches with:  10 <= min GQ <  15
 58  searches with:  15 <= min GQ <  20
  4  searches with:  20 <= min GQ <  25
671  searches with:  25 <= min GQ <  30
  0  searches with:  30 <= min GQ <  35
  0  searches with:  35 <= min GQ <  40
  0  searches with:  40 <= min GQ <  45
  0  searches with:  45 <= min GQ <  50
```

Reply:
So i'm already going to be restricting how gq/ab filtering works in new search to use steps of 5 instead of 1, so I didn't want to *also* restrict the range of filtering that is allowed. It just felt like too many changes at once. Also, the Engle index is still smaller than it was before, so at least this won't use more space for now. That said, I think in the future I will remove some of these bins, but it is much easier to remove than to add

Reply:
Also, not sure if it's worth optimizing this but the 95 to 100 GQ bin will have the most samples. 

![image](https://user-images.githubusercontent.com/6240170/53431379-699d5980-39be-11e9-9427-2c70a755191f.png)
(from  https://gnomad.broadinstitute.org/variant/X-31139927-G-A)

Reply:
at this point I'm not doing more optimizing. it works for engle in reasonable time and space. I'm open to optimizations in the future

Reply:
That's great!! congrats!

Reply:
Are bins for `0 <= gq < 5` and/or `0 < ab < 0.05` needed?

Reply:
No, filtering for > 0 has no effect so we ignore it